diff -Naur nvidia-settings-465.19.01.old/doc/nvidia-settings.desktop nvidia-settings-465.19.01/doc/nvidia-settings.desktop
--- nvidia-settings-465.19.01.old/doc/nvidia-settings.desktop	2021-04-09 18:19:30.141261041 +0200
+++ nvidia-settings-465.19.01/doc/nvidia-settings.desktop	2021-04-09 18:19:38.321401370 +0200
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=NVIDIA X Server Settings
 Comment=Configure NVIDIA X Server Settings
-Exec=__UTILS_PATH__/nvidia-settings
-Icon=__PIXMAP_PATH__/nvidia-settings.png
-Categories=__NVIDIA_SETTINGS_DESKTOP_CATEGORIES__
+Exec=nvidia-settings
+Icon=nvidia-settings
+Categories=System;Settings
+Keywords=nvidia;vulkan;opengl;driver
 
 # Translation by Marcin Mikołajczak
 Name[pl]=Ustawienia serwera X NVIDIA
