diff -Naur nvidia-settings-515.43.04.old/src/libXNVCtrl/utils.mk nvidia-settings-515.43.04/src/libXNVCtrl/utils.mk
--- nvidia-settings-515.43.04.old/src/libXNVCtrl/utils.mk	2022-05-12 13:22:02.541971879 +0200
+++ nvidia-settings-515.43.04/src/libXNVCtrl/utils.mk	2022-05-12 13:26:02.606316574 +0200
@@ -89,7 +89,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
diff -Naur nvidia-settings-515.43.04.old/utils.mk nvidia-settings-515.43.04/utils.mk
--- nvidia-settings-515.43.04.old/utils.mk	2022-05-12 13:22:02.504971209 +0200
+++ nvidia-settings-515.43.04/utils.mk	2022-05-12 13:26:02.607316592 +0200
@@ -89,7 +89,7 @@
 
 INSTALL               ?= install
 INSTALL_BIN_ARGS      ?= -m 755
-INSTALL_LIB_ARGS      ?= -m 644
+INSTALL_LIB_ARGS      ?= -m 755
 INSTALL_DOC_ARGS      ?= -m 644
 
 M4                    ?= m4
