diff -Naur nvidia-settings-515.43.04.old/src/libXNVCtrl/Makefile nvidia-settings-515.43.04/src/libXNVCtrl/Makefile
--- nvidia-settings-515.43.04.old/src/libXNVCtrl/Makefile	2022-05-12 13:22:02.540971860 +0200
+++ nvidia-settings-515.43.04/src/libXNVCtrl/Makefile	2022-05-12 13:25:23.179603052 +0200
@@ -51,7 +51,9 @@
 .PHONY: clean
 
 all: $(LIBXNVCTRL)
+all: $(LIBXNVCTRL_SHARED)
 
 clean:
 	rm -rf $(LIBXNVCTRL) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d
+	rm -f $(LIBXNVCTRL_SHARED) $(LIBXNVCTRL_SHARED).*
diff -Naur nvidia-settings-515.43.04.old/src/libXNVCtrl/xnvctrl.mk nvidia-settings-515.43.04/src/libXNVCtrl/xnvctrl.mk
--- nvidia-settings-515.43.04.old/src/libXNVCtrl/xnvctrl.mk	2022-05-12 13:22:02.541971879 +0200
+++ nvidia-settings-515.43.04/src/libXNVCtrl/xnvctrl.mk	2022-05-12 13:25:23.214603685 +0200
@@ -37,7 +37,8 @@
 XNVCTRL_CFLAGS ?=
 
 
-LIBXNVCTRL = $(OUTPUTDIR)/libXNVCtrl.a
+LIBXNVCTRL = libXNVCtrl.a
+LIBXNVCTRL_SHARED = libXNVCtrl.so
 
 LIBXNVCTRL_SRC = $(XNVCTRL_DIR)/NVCtrl.c
 
@@ -47,3 +48,9 @@
 
 $(LIBXNVCTRL) : $(LIBXNVCTRL_OBJ)
 	$(call quiet_cmd,AR) ru $@ $(LIBXNVCTRL_OBJ)
+
+$(LIBXNVCTRL_SHARED): $(LIBXNVCTRL_OBJ)
+	$(RM) $@ $@.*
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.0.0.0 $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.0.0.0 $@.0
+	ln -s $@.0 $@
diff -Naur nvidia-settings-515.43.04.old/src/Makefile nvidia-settings-515.43.04/src/Makefile
--- nvidia-settings-515.43.04.old/src/Makefile	2022-05-12 13:25:11.134385034 +0200
+++ nvidia-settings-515.43.04/src/Makefile	2022-05-12 13:25:23.215603704 +0200
@@ -281,7 +281,11 @@
 
 all: $(NVIDIA_SETTINGS) $(GTK2LIB) $(GTK3LIB) $(WAYLANDLIB)
 
-install: NVIDIA_SETTINGS_install NVIDIA_GTKLIB_install WAYLAND_LIB_install
+install: NVIDIA_SETTINGS_install NVIDIA_GTKLIB_install WAYLAND_LIB_install NVIDIA_XNVCTRL_install
+
+NVIDIA_XNVCTRL_install:
+	$(MKDIR) $(LIBDIR)
+	cp -a $(LIBXNVCTRL_SHARED)* $(LIBDIR)
 
 NVIDIA_GTKLIB_install: $(GTK2LIB) $(GTK3LIB)
 	$(MKDIR) $(LIBDIR)
@@ -302,26 +306,26 @@
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
-$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL)
+$(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL_SHARED)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL) $(LIBS)
+	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL_SHARED) $(LIBS)
 
 $(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
-$(GTK2LIB).unstripped: $(LIBXNVCTRL) $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
+$(GTK2LIB).unstripped: $(LIBXNVCTRL_SHARED) $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
 	$(call quiet_cmd,LINK) -shared $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK2LIB_SONAME) \
 	    $(GTK2_OBJS) $(IMAGE_OBJS) $(XCP_OBJS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK2_LIBS)
+	    $(LIBXNVCTRL_SHARED) $(LIBS) $(GTK2_LIBS)
 
 ifdef BUILD_GTK3LIB
 $(eval $(call DEBUG_INFO_RULES, $(GTK3LIB)))
-$(GTK3LIB).unstripped: $(LIBXNVCTRL) $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
+$(GTK3LIB).unstripped: $(LIBXNVCTRL_SHARED) $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
 	$(call quiet_cmd,LINK) -shared $(CFLAGS) $(LDFLAGS)  $(BIN_LDFLAGS) \
 	    -o $@ \
 	    -Wl,-soname -Wl,$(GTK3LIB_SONAME) \
 	    $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK3_LIBS)
+	    $(LIBXNVCTRL_SHARED) $(LIBS) $(GTK3_LIBS)
 endif
 
 ifdef BUILD_WAYLANDLIB
@@ -344,7 +348,7 @@
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
 		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
 		$(WAYLANDLIB) $(WAYLANDLIB_DIR) \
-		$(IMAGE_HEADERS) $(LIBXNVCTRL)
+		$(IMAGE_HEADERS) $(LIBXNVCTRL_SHARED)
 
 $(foreach src,$(GTK_SRC), \
     $(eval $(call DEFINE_OBJECT_RULE_WITH_DIR,TARGET,$(src),$(GTK2LIB_DIR))))
