diff -Naur nvidia-settings-600.07.old/src/libXNVCtrl/utils.mk nvidia-settings-600.07/src/libXNVCtrl/utils.mk
--- nvidia-settings-600.07.old/src/libXNVCtrl/utils.mk	2026-01-21 12:45:30.214577005 +0000
+++ nvidia-settings-600.07/src/libXNVCtrl/utils.mk	2026-01-21 16:55:36.794702098 +0000
@@ -600,13 +600,21 @@
   LD_TARGET_EMULATION_FLAG = $(if $(LD_IS_LLD), -m $(LD_TARGET_EMULATION_FLAG_$(TARGET_OS)_$(TARGET_ARCH)))
 endif
 
+# ld on Ubuntu 26.04 crashes when linking binaries inside object, switch to objcopy:
+
+ifeq ($(TARGET_ARCH),x86_64)
+OBJCOPY_BIN_FLAGS = -O elf64-x86-64 -B i386:x86-64
+else ifeq ($(TARGET_ARCH),aarch64)
+OBJCOPY_BIN_FLAGS = -O elf64-littleaarch64 -B aarch64
+endif
+
 define READ_ONLY_OBJECT_FROM_FILE_RULE
   $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
 	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
 	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
-	    $$(LD_TARGET_EMULATION_FLAG) \
-	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
+	$$(call quiet_cmd_no_at,OBJCOPY) -I binary \
+	    $$(OBJCOPY_BIN_FLAGS) \
+	    $$(notdir $(1)) $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
 	$$(call quiet_cmd,OBJCOPY) \
 	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \
 	    $$@
diff -Naur nvidia-settings-600.07.old/utils.mk nvidia-settings-600.07/utils.mk
--- nvidia-settings-600.07.old/utils.mk	2026-01-21 12:45:30.229576790 +0000
+++ nvidia-settings-600.07/utils.mk	2026-01-21 16:55:45.385629899 +0000
@@ -600,13 +600,21 @@
   LD_TARGET_EMULATION_FLAG = $(if $(LD_IS_LLD), -m $(LD_TARGET_EMULATION_FLAG_$(TARGET_OS)_$(TARGET_ARCH)))
 endif
 
+# ld on Ubuntu 26.04 crashes when linking binaries inside object, switch to objcopy:
+
+ifeq ($(TARGET_ARCH),x86_64)
+OBJCOPY_BIN_FLAGS = -O elf64-x86-64 -B i386:x86-64
+else ifeq ($(TARGET_ARCH),aarch64)
+OBJCOPY_BIN_FLAGS = -O elf64-littleaarch64 -B aarch64
+endif
+
 define READ_ONLY_OBJECT_FROM_FILE_RULE
   $$(OUTPUTDIR)/$$(notdir $(1)).o: $(1)
 	$(at_if_quiet)$$(MKDIR) $$(OUTPUTDIR)
 	$(at_if_quiet)cd $$(dir $(1)); \
-	$$(call quiet_cmd_no_at,LD) -r -z noexecstack --format=binary \
-	    $$(LD_TARGET_EMULATION_FLAG) \
-	    $$(notdir $(1)) -o $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
+	$$(call quiet_cmd_no_at,OBJCOPY) -I binary \
+	    $$(OBJCOPY_BIN_FLAGS) \
+	    $$(notdir $(1)) $$(OUTPUTDIR_ABSOLUTE)/$$(notdir $$@)
 	$$(call quiet_cmd,OBJCOPY) \
 	    --rename-section .data=.rodata,contents,alloc,load,data,readonly \
 	    $$@
